flink-clients/src/main/java/org/apache/flink/client/program/ContextEnvironment.java

```diff
@@ -91,25 +82,27 @@ public JobExecutionResult execute(String jobName) throws Exception {
 			}
 
 			jobExecutionResult = jobExecutionResultFuture.get();
+			jobExecutionResult.printResult();
 		} else {
 			jobExecutionResult = new DetachedJobExecutionResult(jobClient.getJobID());
 		}
 
-		setJobExecutionResult(jobExecutionResult);
 		return jobExecutionResult;
 	}
 
-	private void verifyExecuteIsCalledOnceWhenInDetachedMode() {
+	@Override
+	public JobClient executeAsync(String jobName) throws Exception {
+		verifyExecuteIsCalledOnceWhenInDetachedMode();
+		return super.executeAsync(jobName);
+	}
+
+	public void verifyExecuteIsCalledOnceWhenInDetachedMode() {
```

Comment:
I think we actually don't need these "verifications" anymore, with the new architecture where the Executor is called inside the `execute()` method and where we don't actually "hijack" the method anymore, we can actually have multiple execute() calls. We can address that in a follow-up, though.

Reply:
I agree and can open a ticket for this, even add a tiny commit on this thread. However, I'm a bit unclear about your idea as comment below.